### PR TITLE
TS-4731: Changes return codes for IsInternal APIs

### PR DESCRIPTION
**Pull request by zwoop**
This changes the following APIs:

tsapi int TSHttpTxnIsInternal(TSHttpTxn txnp);
tsapi int TSHttpSsnIsInternal(TSHttpSsn ssnp);

In addition, this also removes the following APIs, that were previously
deprecated:

tsapi TS_DEPRECATED TSReturnCode TSHttpIsInternalRequest(TSHttpTxn txnp);
tsapi TS_DEPRECATED TSReturnCode TSHttpIsInternalSession(TSHttpSsn ssnp);

### CODE REVIEW

**Comment by zwoop** [comment]
Whomever is reviewing: Please make sure I got all the conditionals right, with the correct boolean logic.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/753/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/649/ for details.

File: doc/developer-guide/api/functions/TSHttpTxnIsInternal.en.rst [hunk 1]
```diff
@@ -41,8 +41,8 @@ was originated within Traffic Server.
 Return Values
 =============
 
-Both these APIs returns a :type:`TSReturnCode`, indicating whether the
-object was internal (:data:`TS_SUCCESS`) or not (:data:`TS_ERROR`).
+Both these APIs returns a :type:`int`, indicating whether the
```
Comment (jpeach) [inline doc/developer-guide/api/functions/TSHttpTxnIsInternal.en.rst]: s/returns/return/

File: doc/developer-guide/api/functions/TSHttpTxnIsInternal.en.rst [hunk 2]
```diff
@@ -41,8 +41,8 @@ was originated within Traffic Server.
...
+object was internal (:data:`1`) or not (:data:`0`).
```
Comment (jpeach) [inline doc/developer-guide/api/functions/TSHttpTxnIsInternal.en.rst]: s/object/transaction/

**Comment by jpeach** [comment]
👍 

Consider a heads-up email to dev@ since this change is likely to break 3rd party plugins.

**Comment by jpeach** [comment]
Rebase!

**Comment by bryancall** [comment]
👍 - Looks good, but needs an update

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/685/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/789/ for details.

**Pull request merged**